base/boot.jl

```diff
@@ -269,9 +269,9 @@ Void() = nothing
 
 immutable VecElement{T}
     value::T
-    VecElement{T}(value::T) where T = new(value) # disable converting constructor in Core
+    VecElement{T}(value::T) where {T} = new(value) # disable converting constructor in Core
 end
-VecElement(arg::T) where T = VecElement{T}(arg)
+VecElement(arg::T) where {T} = VecElement{T}(arg)
```

Comment:
It's a pretty common complaint, e.g. looking at the comments in #11310.

Currently the syntax

```
function Complex{T}(x::T) where T
```

unambiguously defines a constructor for `Complex{T}`. But if you remove the `where T`, the `{T}` is interpreted as introducing static parameters, so instead it defines a constructor for `Complex`. This is confusing. Alternatives like `Complex{T}{T}(x)` or `Complex{T}(x){T}` seem more obscure to me.